app/src/main/java/com/my/journeyplanner/presenters/main/MainPresenter.kt

```diff
@@ -18,18 +18,20 @@ class MainPresenter(private val view: MainContract.View, private val interactors
     }
 
     override fun onPlanMyJourneyClicked() {
-        call = interactors.getJourneyResults(
+        call = retrofitApi.getJourneyResults(
             view.getFromLocation().text.toString(),
             view.getToLocation().text.toString()
         )
-        interactors.enqueue(call!!)
+        retrofitApi.enqueue(call!!)
+
+        view.showResult(retrofitApi.journeyPlannerResultDomainModel.toString())
```

Comment:
how did you ensure that enqueue (which is async) has returned before this line (as you're accessing result)